api/db-cdr/changelog/db.changelog-5.xml

```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.0.xsd
+        http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd">
+  <changeSet author="brianfreeman" id="changelog-5" runAlways="true">
+
+    <sql dbms="mysql"
+         endDelimiter=";"
+         splitStatements="true"
+         stripComments="true">update criteria t1
+      inner join criteria t2 on (t1.parent_id = t2.sort_order and t1.type like 'DEMO%' and t2.type like 'DEMO%')
```

Comment:
I don't understand this SQL. Is "sort_order" an external ID? If so, why is it named "sort_order"? Why not just have it be the ID for the table?

Reply:
So the trees are ordered by there original id in the vanderbilt tables. So i felt like that was fragile since we may be generating csv every so often and merging the tables together. So i created a sort_order that the table needs to be order by and we can generate any id we want for the id. That way the csv's can be out of order and we don't care.

Reply:
this sql updates the parent_id to the newly generate id since all criteria will live the same table. This is because the csv files i have are coming from them being in separate tables

Reply:
OK, so sort_order is both an external key (within the scope of a given type) and the order in which we want records to appear in the UI, and parent_id starts life as a foreign key to that and winds up being a foreign key to the criteria table... is that right?

A couple thoughts:

- If we guaranteed that rows appeared in sort_order order in the CSV file, you could probably drop the sort_order column and just order by ID. (Though having a separate sort_order column is not necessarily a bad thing.)
- You should probably add a foreign key constraint on parent_id once everything is done. This will wind up creating an index on parent_id. You might want to consider using a different column for the external FK during the import and dropping it when finished. 
- You probably should document what sort_order means in the Java.